### PR TITLE
tns-core-modules + d.ts = ☺☺☺

### PR DESCRIPTION
**Pull request by hdeshev**
Include d.ts files in tns-core-modules package.
    
- Generate a combined tns-core-modules.d.ts file for user projects. (Will
be included in tsconfig.json files by the `tns` tool automagically.)
- Test the integrity of the generated combined file by compiling it.
- Remove all traces of the obsoleted tns-definitions package.

/cc @ErjanGavalji @tailsu @ligaz

### CODE REVIEW

**Comment by hdeshev** [comment]
And here's how client apps' tsconfig.json should look like:

```json
    "filesGlob": [
        "src/**/*.ts",
        "node_modules/tns-core-modules/tns-core-modules.d.ts"
    ],
    "files": [
        ...
        "node_modules/tns-core-modules/tns-core-modules.d.ts"
    ]

```

File: gruntfile.js [hunk 1]
```diff
@@ -512,11 +492,23 @@ module.exports = function(grunt) {
...
         ]);
-        grunt.registerTask("distribute-definition-files", [
-                "copy:definitionFiles"
-        ]);
+        grunt.registerTask("generate-tns-core-modules-dts", function() {
+            var dtsFiles = grunt.file.expand({cwd: localCfg.outModulesDir}, ['**/*.d.ts']);
+            dtsFiles.sort()
+
+            var dtsLines = [];
+            dtsFiles.forEach(function(dtsFile) {
```
Comment (ligaz) [inline gruntfile.js]: You can use `map(...)` here.

**Comment by enchev** [comment]
:+1

**Comment by ErjanGavalji** [comment]
:+1:

**Comment by vakrilov** [comment]
:100:

File: gruntfile.js (hunk 1 above)
Comment (hdeshev) [inline gruntfile.js]: Fixed!

**Comment by nsndeck** [comment]
:+1:

**Comment by tailsu** [comment]
The package.json should have a "typings" property with the path to the .d.ts file.

**Comment by hdeshev** [comment]
Coming in PR #1030. Forgot to add it here.

Dean

**Pull request merged**